lib/turn.rb

```diff
@@ -1,32 +1,52 @@
 class Turn
-  attr_reader :player, :board
+  attr_reader :player, :board, :columns
 
   def initialize(player, board)
     @player = player
     @board = board
+    @columns = ["a", "b", "c", "d", "e", "f", "g"]
   end
 
   def get_move
-    columns = ["a", "b", "c", "d", "e", "f", "g"]
-    move = validate_move(gets.chomp.downcase, columns)
-    @board.board_grid.reverse.each do |row|
-      if row[(columns.index(move))].state == "."
-        set_cell(@board.board_grid.index(row), columns.index(move))
-        break
-      end
-    end
+    move = validate_move(gets.chomp.downcase, @columns)
+    find_lowest_cell_in_column(move, @columns)
   end
-
+  
   def validate_move(move, valid_columns)
     while !valid_columns.include?(move) || column_is_full?(move, valid_columns)
       puts "Oops, that's an invalid move, or a full column. Please select column A-G!"
       move = gets.chomp.downcase
     end
     move
   end
+  
+  def get_computer_move
+    computer_move = @columns.sample
+    while validate_cpu_move(computer_move, columns) == "invalid"
+      computer_move = @columns.sample
+    end
+    find_lowest_cell_in_column(computer_move, @columns)
+  end
+
+  def validate_cpu_move(move, valid_columns)
+    if column_is_full?(move, valid_columns)
+      return "invalid"
+    else 
+      move
+    end
+  end
+
+  def find_lowest_cell_in_column(move, valid_columns)
+    @board.board_grid.reverse.each do |row|
+      if row[(valid_columns.index(move))].state == "."
+        set_cell(@board.board_grid.index(row), valid_columns.index(move))
```

Comment:
So we now have somewhat of a tangled web all leading to #set_cell...we should probably refactor later down the road to make find_lowest_cell_in_column complete standalone, rather than it leading directly to the piece placement. If I called find_lowest_cell_in_column, I would probably expect a return value rather than it just straight up putting the piece down?

That being said, 100% works for now and I like how you pulled it out!